docs: add doc for workforce

WHALEEYE opened the pull request:
## Description

Describe your changes in detail.

## Motivation and Context

 `close #771 `

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

Wendong-Fan commented on docs/key_modules/workforce.md:
```diff
@@ -0,0 +1,255 @@
+
+When all the subtasks are finished, their results will be composed into a single, comprehensive result and then get updated into the main task instance and returned to the user.
+
+## A Full Demo of Workforce
+
+Below is a comprehensive example of the using of workforce.
```
IMO we can remove this part as it's a little bit redundant, WDT?

WHALEEYE replied:
You mean we just keep the step-by-step instructions?

Wendong-Fan commented on docs/key_modules/workforce.md:
````diff
@@ -0,0 +1,255 @@
+Finally, we can get the result of the task by checking `task.result`.
+
+```python
+print('Final Result of Origin task:\n', task.result)
+```
````
add the output from print would be better

Wendong-Fan reviewed (commented):
thanks @WHALEEYE ! we can also add the architecture diagram to this part, also mentioned we are going to develop and release 2nd version for better workforce

WHALEEYE commented:
@Wendong-Fan Here are some problems I found after I make changes according to the review:
- Deleted the full demo since it's kind of redundant. But I left an empty link in the step-by-step guide section, where I think we can put the full demo.
- Added the diagram of the workflow into the workflow section, but I don't know where to put the image file. Should we create a new folder to store these files?
- Following the diagram, I think it's kind of too large (and too much information), but I'm not sure what to put inside the diagram. Maybe we can have a discussion on this.

Wendong-Fan commented:
> @Wendong-Fan Here are some problems I found after I make changes according to the review:
> 
> * Deleted the full demo since it's kind of redundant. But I left an empty link in the step-by-step guide section, where I think we can put the full demo.
> * Added the diagram of the workflow into the workflow section, but I don't know where to put the image file. Should we create a new folder to store these files?
> * Following the diagram, I think it's kind of too large (and too much information), but I'm not sure what to put inside the diagram. Maybe we can have a discussion on this.

Thanks @WHALEEYE ! I added one commit here: https://github.com/camel-ai/camel/pull/786/commits/1f6bae999defff9df1ce248cfa14059c4d14ed20
